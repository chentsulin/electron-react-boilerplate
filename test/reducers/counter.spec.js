import {
  counterReducer,
  increment,
  decrement
} from '../../app/reducers/counter';

describe('reducers', () => {
  describe('counter', () => {
    it('should handle initial state', () => {
      expect(counterReducer(undefined, {})).toMatchSnapshot();
    });

    it('should handle increment', () => {
      expect(counterReducer(1, { type: increment })).toMatchSnapshot();
    });

    it('should handle decrement', () => {
      expect(counterReducer(1, { type: decrement })).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(counterReducer(1, { type: 'unknown' })).toMatchSnapshot();
    });
  });
});
