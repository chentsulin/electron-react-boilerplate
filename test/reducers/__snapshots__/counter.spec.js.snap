// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers counter should handle counter/decrement 1`] = `2`;

exports[`reducers counter should handle counter/increment 1`] = `0`;

exports[`reducers counter should handle decrement 1`] = `0`;

exports[`reducers counter should handle increment 1`] = `2`;

exports[`reducers counter should handle initial state 1`] = `0`;

exports[`reducers counter should handle unknown action type 1`] = `1`;
